### DISCUSSION TITLE
Integration of Next.js with Azure Active Directory

### DISCUSSION THREAD

**Discussion by jackHedaya** [vercel/next.js#15210] [Help]
Hey!

I'm trying to integrate my Next.js app with Azure AD but am having trouble because all of the examples I'm finding online require fine-tune support with express objects to run. There seems to be a need for support for express middlewares like passport or express-session to work. Anyone have any ideas on which direction to go? In the ideal world, I'd be able to set up AD in an API endpoint and redirect unauthorized requests in gSS part of my pages.

Thanks!

**Comment by subhashdoddaga1907** [comment]
@jackHedaya can u explain how you configured azure ad in nextJs

**Reply by jackHedaya** [reply]
Haven’t gotten to it yet as there were other more valuable parts of my project. Please let me know if you end up using it though.

**Comment by kvnify** [comment]
I've just followed the examples out there using `passport` and `express` and the custom server documentation that `next` provides. Seems to be working so far, but I'm still just in the initial stages of putting it all together. Not sure what authentication strategy you are using, but I was able to get the Open ID flow to work in my case.

**Reply by jackHedaya** [reply]
Have any repo example you'd be willing to share?

**Reply by kvnify** [reply]
Nope. Sorry. It’s in a private repo.
